source/Personality.cpp

```diff
@@ -148,11 +152,31 @@ void Personality::Load(const DataNode &node)
 		if(child.Token(0) == "confusion")
 		{
 			if(add || remove)
-				child.PrintTrace("Cannot \"" + node.Token(0) + "\" a confusion value:");
-			else if(child.Size() < 2)
-				child.PrintTrace("Skipping \"confusion\" tag with no value specified:");
-			else
+			{
+				child.PrintTrace("Cannot \"" + node.Token(0) + "\" a confusion node:");
+				continue;
+			}
+			// Accept the old method of defining confusion for backwards compatibility.
+			if(child.Size() == 2)
```

Comment:
We usually aren't strict about expecting an exact number of tokens.
```suggestion
			if(child.Size() >= 2)
```